components/layout_2020/dom_traversal.rs

```diff
@@ -39,22 +40,30 @@ impl<Node> NodeAndStyleInfo<Node> {
         style: ServoArc<ComputedValues>,
     ) -> Self {
         Self {
-            node,
+            node: Some(node),
             pseudo_element_type: Some(pseudo_element_type),
             style,
         }
     }
 
     pub(crate) fn new(node: Node, style: ServoArc<ComputedValues>) -> Self {
         Self {
-            node,
+            node: Some(node),
             pseudo_element_type: None,
             style,
         }
     }
 }
 
 impl<Node: Clone> NodeAndStyleInfo<Node> {
+    pub(crate) fn new_anonymous(&self, style: ServoArc<ComputedValues>) -> Self {
+        Self {
+            node: None,
+            pseudo_element_type: self.pseudo_element_type,
+            style,
+        }
+    }
+
     pub(crate) fn new_replacing_style(&self, style: ServoArc<ComputedValues>) -> Self {
```

Comment:
Should all callers of this use `new_anonymous`?

Reply:
I chose to keep it for a couple callers which expect things to be inherited or which are creating list markers. There's a larger refactor we should do to clean this up and we need to implement the `::marker` pseudo, but these are pretty intrusive things that should be done in a larger project.